### PR TITLE
Research/improve dx integration

### PR DESCRIPTION
**Pull request by pgarciaegido**
Here I make significant changes to the way our SDK is integrated. Here are the main improvements, and the nitty gritty on how they work.

### How?
I have moved the state from the providers all the way down to the button. State is handled in two different hooks:
- `packages/ui/react-ui/src/hooks/useCrossMintStatus.ts`: Here we handle the status of the client, from the client-id provided.
- `packages/ui/react-ui/src/hooks/useCrossMintModal.ts` : Here we handle the state of the popup window.

We used to have a React component that created an overlay on the page, and its state was handled by a context at the provider level. Now, we manually create and append the div to the body when the user clicks on the crossmint button, and remove it when popup is closed. And actually, this is now optional (true by default). See more at the bottom.

### How?
Using a CSS in JS library (styled-components) that would bind the styles to the bundled output files. This actually simplifies a lot the customization of the button in the user side. We can leverage css specification and simply use `button.<custom-class>` in order to remove the need to use `!important`s. (See `packages/starter/nextjs-starter/styles/globals.css`)

# New options to the button

Now the button is taking also the `clientId` prop, but also the `development` and some others (check packages/ui/react-ui/package/src/CrossMintButton.tsx interface). 

I have added 3 more (optionals):

- `showOverlay` True by default. Will show an overlay when the popup window is open and will remove it when its closed (like it works now). If false, button text will continue to change but overlay wont be shown.
- `crossmintOpened` This expects a callback function that will be called everytime the popup window opens. This is handy for the devs to want to show a custom level or just to be able to track the state of the window.
- `crossmintClosed` This expects a callback function that will be called everytime the popup window closes.

# Downsides of this approach

- I would say a downside is using and external library for the css in js. Now our g-ziped bundle size is around ~13kb which is not crazy, but certainly could be much lower. I might need to do a little research on finding another ligther css in js lib.
- Not having a provider (crossmint global state available) and having to use callbacks to control the state of the button might be annoying to some more experienced devs that want to be able to check on the crossmint button anywhere on their app. There is a trade off here.

BTW guys, you see that I have created a pack script and linked the package manually on the next.js starter... I dont know if there is any other way you guys do that (maybe we could use Storybook for this?), but that will be removed if this goes forward.

⚠️ This needs to be tested more in detail before merging!

Thanks a lot!

### CODE REVIEW

File: packages/ui/react-ui/package.json
Comment (pgarciaegido) [inline packages/ui/react-ui/package.json:30]: Here Im bulding the package locally in order to test in the nextjs starter.

**Review by mPaella — CHANGES_REQUESTED** [review]

File: packages/starter/nextjs-starter/pages/index.tsx
Comment (alfonso-paella) [reply packages/starter/nextjs-starter/pages/index.tsx:18]: We should do https://github.com/CrossMint/crossmint-client-sdk/issues/14 to fix this for good

Comment (alfonso-paella) [reply packages/starter/nextjs-starter/pages/index.tsx:18]: And modify this to say "<YOUR_CLIENT_ID>"

File: packages/starter/nextjs-starter/pages/index.tsx
Comment (alfonso-paella) [reply packages/starter/nextjs-starter/pages/index.tsx:19]: +1. Maybe we can have a non customized button, and then a separate customized one? And add a comment in the cusotmized one explaining how it works

Comment (alfonso-paella) [reply packages/starter/nextjs-starter/pages/index.tsx:19]: Also I would rename the classname to "my-custom-style-crossmint-button" or something like that, super obvious

File: packages/ui/react-ui/src/CrossMintButton.tsx
Comment (alfonso-paella) [reply packages/ui/react-ui/src/CrossMintButton.tsx:23]: If these are new parameters that didn't exist before, can we take them out of this PR and send a different PR for them? OR file an issue to track

I'd want more time to define the API surface properly, but didn't want to block this PR from going out in the meantime

File: packages/ui/react-ui/src/hooks/useCrossMintModal.ts
Comment (alfonso-paella) [inline packages/ui/react-ui/src/hooks/useCrossMintModal.ts:43]: This file doesn't seem to have been run through prettier

File: packages/ui/react-ui/src/hooks/useCrossMintModal.ts
Comment (alfonso-paella) [inline packages/ui/react-ui/src/hooks/useCrossMintModal.ts:112]: Can we make this a console.error and have it be more actionable for the developer? Can they do anything to fix this

**Review by alfonso-paella — CHANGES_REQUESTED** [review]

File: packages/ui/react-ui/package.json
Comment (pgarciaegido) [reply packages/ui/react-ui/package.json:30]: yep yep, will delete 😄

File: packages/starter/nextjs-starter/pages/index.tsx
Comment (pgarciaegido) [reply packages/starter/nextjs-starter/pages/index.tsx:18]: Sorry 😅

File: packages/starter/nextjs-starter/pages/index.tsx
Comment (pgarciaegido) [reply packages/starter/nextjs-starter/pages/index.tsx:19]: I would not give it up tbh. It makes the most sense to me at least. How about we just remove it from the examples and snippets, and maybe add a little comment to it in the very bottom of the docs?

File: packages/ui/react-ui/src/CrossMintButton.tsx
Comment (pgarciaegido) [reply packages/ui/react-ui/src/CrossMintButton.tsx:23]: Ok @alfonso-paella lets do that 😄

File: packages/ui/react-ui/src/hooks/useCrossMintModal.ts
Comment (pgarciaegido) [reply packages/ui/react-ui/src/hooks/useCrossMintModal.ts:43]: We dont have prettier set up in this repo I believe

File: packages/ui/react-ui/src/hooks/useCrossMintModal.ts
Comment (pgarciaegido) [reply packages/ui/react-ui/src/hooks/useCrossMintModal.ts:112]: sure! @mPaella anything comes to mind?

File: packages/ui/react-ui/src/CrossMintButton.tsx
Comment (pgarciaegido) [reply packages/ui/react-ui/src/CrossMintButton.tsx:23]: Created https://github.com/CrossMint/crossmint-client-sdk/issues/34

File: packages/starter/nextjs-starter/pages/index.tsx
Comment (mPaella) [reply packages/starter/nextjs-starter/pages/index.tsx]: Most good devs will just look [here](https://github.com/CrossMint/crossmint-client-sdk/tree/main/packages/ui/react-ui), or look at the ts type and see they can use this

But seems like a lot of the people we r dealing with are not good devs

Comment (pgarciaegido) [reply packages/starter/nextjs-starter/pages/index.tsx]: Yeah... I mean to keep the prop as `className`.

Obviously, it needs to be gone from the example haha

Comment (alfonso-paella) [reply packages/starter/nextjs-starter/pages/index.tsx]: Yeah the prop seems fine, my comment (and I think Maxwell's) was about keeping it in the main example

File: packages/ui/react-ui/src/hooks/useCrossMintModal.ts
Comment (alfonso-paella) [reply packages/ui/react-ui/src/hooks/useCrossMintModal.ts:43]: Oops, we should add it then. Mind running it manually for now?

**Comment by pgarciaegido** [comment]
onOpen and onClose events already deleted 👍

File: packages/ui/react-ui/src/hooks/useCrossMintModal.ts
Comment (alfonso-paella) [reply packages/ui/react-ui/src/hooks/useCrossMintModal.ts]: If it's not obvious let's leave a TODO and do in another task

File: packages/starter/nextjs-starter/next.config.js
Comment (alfonso-paella) [inline packages/starter/nextjs-starter/next.config.js:3]: Run prettier in this file

**Comment by alfonso-paella** [comment]
Mind adding back the onclick ? Until we figure out what to do with the onopen and onclose

**Review by alfonso-paella — APPROVED** [review]
I'm good pending the couple unresolved comments. Once you address those feel free to merge if you feel comfortable

**Pull request merged**
